run/main.go

```diff
@@ -41,8 +42,17 @@ func Run() int {
 			printStackTo(os.Stdout, "PRINTING STACK ON EXIT (STARTUP ERROR)")
 		}
 
+		// Trigger shutdown and wait for it to complete.
 		_ = modules.Shutdown()
-		return modules.GetExitStatusCode()
+		exitCode := modules.GetExitStatusCode()
+
+		// Return the exit code, if it was set.
+		if exitCode > 0 {
+			return exitCode
+		}
+
+		// Otherwise, return a default 1.
+		return 1
```

Comment:
That is somehow weird because you cannot distinguish between a returned `1` being an error from the module system or not.

Reply:
Granted. Maybe we should document this behavior?
`1` for me is a default error. If I'd specify one, I'd always use a different one anyway - that led me to do it that way.